Automattic/jetpack
# Slideshows: don't attempt to use the block when not available

Fixes #38742

## Proposed changes:

Folks can disable Jetpack blocks on the whole site. While that impact blocks' availability in the block editor, it also impacts parts of the codebase where we attempt to use functions and classes within the `Automattic\Jetpack\Extensions` namespace.

The old classic slideshow is a good example of that: when used with the AMP plugin, it attempts to display a Slideshow block instead of the classic slideshow in AMP views.

In this PR, we do not attempt to use the block when blocks are not available.

> [!NOTE]
> ~This is only a partial fix. Since some AMP views do not load JavaScript, and since this PR reverts to using the classic JavaScript-powered slideshow shortcode when the block isn't available, the slideshow will not be displayed properly.~
> ~**An alternative solution could be to require the slideshow block file in this scenario.**~
> I have now pushed 1f9c526213462d4253876608d235046ec7d84307 to address that.

### Other information:


## Jetpack product discussion

* p1722963449890599-slack-CBG1CP4EN

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

1. New site running Jetpack with this branch.
2. Shortcodes module active.
3. AMP plugin active.
4. In Media > Add New, upload a few images.
5. In Posts > Add New, add a classic block.
6. With the media modal, create a gallery, add the images
7. Pick the slideshow gallery type
8. Publish the post.
9. Visit the post.
10. In the admin menu bar, click on the AMP menu item to view the AMP version of the post.
11. The slideshow block is displayed properly.
12. Now go to Jetpack > Settings > Writing
13. Disable the Jetpack Blocks
14. Refresh the AMP page.
15. You should see no Fatal errors. The block may not be displayed properly depending on your AMP settings on the site though.
